src/api/team_api.rs

```diff
@@ -0,0 +1,176 @@
+//! Pull employee data into accounting and payroll systems with the Team API.
+//!
+//! The Team API allows applications to retrieve a roster of team members registered in the Square
+//! Point of Sale system, which can be useful in payroll and account contexts.
+//!
+//! The Team API is best used in conjunction with the Labor API, where you provide team member IDs
+//! to manage shifts, breaks, and wages.
+
+use crate::{
+    config::Configuration,
+    http::client::HttpClient,
+    models::{
+        errors::ApiError, BulkCreateTeamMembersRequest, BulkCreateTeamMembersResponse,
+        BulkUpdateTeamMembersRequest, BulkUpdateTeamMembersResponse, CreateTeamMemberRequest,
+        CreateTeamMemberResponse, RetrieveTeamMemberResponse, RetrieveWageSettingResponse,
+        SearchTeamMembersRequest, SearchTeamMembersResponse, UpdateTeamMemberRequest,
+        UpdateTeamMemberResponse, UpdateWageSettingRequest, UpdateWageSettingResponse,
+    },
+};
+
+const DEFAULT_URI: &str = "/team-members";
+
+pub struct TeamApi {
+    /// App config information
+    config: Configuration,
+    /// HTTP Client for requests to the Team API endpoints
+    client: HttpClient,
+}
+
+impl TeamApi {
+    /// Instantiates a new `TeamApi`
+    pub fn new(config: Configuration, client: HttpClient) -> Self {
+        Self { config, client }
+    }
+
+    /// Creates a single `TeamMember` object.
+    ///
+    /// The `TeamMember` object is returned on successful creates. You must provide the following
+    /// values in your request to this endpoint:
+    /// * `given_name`
+    /// * `family_name`
+    ///
+    /// Learn about [Troubleshooting the Team
+    /// API](https://developer.squareup.com/docs/team/troubleshooting#createteammember).
+    pub async fn create_team_member(
+        &self,
+        body: &CreateTeamMemberRequest,
+    ) -> Result<CreateTeamMemberResponse, ApiError> {
+        let response = self.client.post(&self.url(), body).await?;
+
+        response.deserialize().await
+    }
+
+    /// Creates multiple `TeamMember` objects.
+    ///
+    /// The created `TeamMember` objects are returned on successful creates. This process is non-
+    /// transactional and processes as much of the request as possible. If one of the creates in the
+    /// request cannot be successfully processed, the request is not marked as failed, but the body
+    /// of the response contains explicit error information for the failed create.
+    ///
+    /// Learn about [Troubleshooting the Team
+    /// API](https://developer.squareup.com/docs/team/troubleshooting#createteammember).
+    pub async fn bulk_create_team_members(
+        &self,
+        body: &BulkCreateTeamMembersRequest,
+    ) -> Result<BulkCreateTeamMembersResponse, ApiError> {
+        let url = format!("{}/bulk-create", self.url());
```

Comment:
These are urls to them, right? It's odd that they aren't using a POST for create and an UPDATE/PATCH for update. But if that's the case then whatever.

Reply:
Yeah, they're using POSTs for both the bulk methods.  The single create method also uses a POST and both single update methods use PUTs.

Reply:
Thanks for checking though.  Made me double check which is a good thing.